lib/_http_agent.js

```diff
@@ -116,10 +117,29 @@ function Agent(options) {
       }
     }
   });
+
+  // Don't emit keylog events unless there is a listener for them.
+  this.on('newListener', maybeEnableKeylog);
 }
 Object.setPrototypeOf(Agent.prototype, EventEmitter.prototype);
 Object.setPrototypeOf(Agent, EventEmitter);
 
+function maybeEnableKeylog(eventName) {
+  if (eventName === 'keylog') {
+    this.removeListener('newListener', maybeEnableKeylog);
+    // Future sockets will listen on keylog at creation.
+    const agent = this;
+    this[kOnKeylog] = function onkeylog(keylog) {
+      agent.emit('keylog', keylog, this);
+    };
+    // Existing sockets will start listening on keylog now.
+    const sockets = Object.values(this.sockets);
+    for (var i = 0; i < sockets.length; i++) {
```

Comment:
```suggestion
    for (let i = 0; i < sockets.length; i++) {
```
I guess you could also DRY this as:
```suggestion
    for (const socket of Object.values(this.sockets)) {
```
And we don't seem to be using primordials for `Object.values()` but maybe we should? No strong opinion.